ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,816 @@
+	}
+
+	// 3. Build price items
+	server, err := getItemPriceId(items, "server", process_key_name.(string))
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	os, err := getItemPriceId(items, "os", osKeyName.(string))
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	ram, err := findMemoryItemPriceId(items, d)
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	portSpeed, err := findNetworkItemPriceId(items, d)
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	monitoring, err := getItemPriceId(items, "monitoring", "MONITORING_HOST_PING")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	if d.Get("tcp_monitoring").(bool) {
+		monitoring, err = getItemPriceId(items, "monitoring", "MONITORING_HOST_PING_AND_TCP_SERVICE")
+		if err != nil {
+			return datatypes.Container_Product_Order{}, err
+		}
+	}
+	// Other common default options
+	priIpAddress, err := getItemPriceId(items, "pri_ip_addresses", "1_IP_ADDRESS")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	pri_ipv6_addresses, err := getItemPriceId(items, "pri_ipv6_addresses", "1_IPV6_ADDRESS")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	remoteManagement, err := getItemPriceId(items, "remote_management", "REBOOT_KVM_OVER_IP")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	vpnManagement, err := getItemPriceId(items, "vpn_management", "UNLIMITED_SSL_VPN_USERS_1_PPTP_VPN_USER_PER_ACCOUNT")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	notification, err := getItemPriceId(items, "notification", "NOTIFICATION_EMAIL_AND_TICKET")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	response, err := getItemPriceId(items, "response", "AUTOMATED_NOTIFICATION")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	vulnerabilityScanner, err := getItemPriceId(items, "vulnerability_scanner", "NESSUS_VULNERABILITY_ASSESSMENT_REPORTING")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	// Define an order object using basic paramters.
+
+	order := datatypes.Container_Product_Order{
+		ContainerIdentifier: sl.String(d.Get("hostname").(string)),
+		Quantity:            sl.Int(1),
+		Hardware: []datatypes.Hardware{{
+			Hostname: sl.String(d.Get("hostname").(string)),
+			Domain:   sl.String(d.Get("domain").(string)),
+		},
+		},
+		Location:  sl.String(strconv.Itoa(*dc.Id)),
+		PackageId: pkg.Id,
+		Prices: []datatypes.Product_Item_Price{
+			server,
+			os,
+			ram,
+			portSpeed,
+			priIpAddress,
+			pri_ipv6_addresses,
+			remoteManagement,
+			vpnManagement,
+			monitoring,
+			notification,
+			response,
+			vulnerabilityScanner,
+		},
+	}
+
+	// Add optional price ids.
+	// Add public bandwidth
+
+	if publicBandwidth, ok := d.GetOk("public_bandwidth"); ok {
```

Comment:
`public_bandwidth` has a default value, hence it will always get one even if user doesn't specify one.

Reply:
Removed unnecessary GetOK